torch/testing/_internal/common_utils.py

```diff
@@ -940,6 +940,35 @@ def get_comparison_dtype(a, b):
 
     return compare_dtype
 
+def compare_with_reference(torch_fn, ref_fn, sample_input):
+    # Compares torch function with reference function for given sample input (object of SampleInput)
+    # Note: only values are compares, type comparison is not done here
+    try:
+        n_inp, n_args, _ = sample_input.numpy()
+    except TypeError:
+        # TODO: This needs a comment
+        return
+    t_inp, t_args, _ = sample_input.splat()
```

Comment:
`torch_fn(sample_input.splat())` returns the following error. I am looking for suggestions on this.

```
======================================================================
ERROR: test_reference_testing_sub_test_subtract_cpu_int64 (__main__.TestOpInfoCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
    result = test_fn(self, *args)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 266, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 627, in wrapper
    fn(*args, **kwargs)
  File "/home/krshrimali/git/krshrimali/pytorch/test/test_ops.py", line 149, in test_reference_testing
    compare_with_reference(op, op.ref, sample_input)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 942, in compare_with_reference
    torch_fn(*sample_input.splat())
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_methods_invocations.py", line 331, in __call__
    return self.op(*args, **kwargs)
TypeError: sub() takes 2 positional arguments but 3 were given

----------------------------------------------------------------------
```

**Note:** Even if we are able to fix this, `n_kwargs` can't be passed and hence they are ignored to be `_`, an example is that `np.add` doesn't have `alpha` argument but `torch.add` has.

@mruberry @pmeier @kshitij12345 - Can you please take a loo at this?

Reply:
Because this is trying to call sub like this:

`sub((a, b, {alpha=.5}))`

when it needs to be calling sub like this:

`sub(a, b, alpha=.5)`

For np.add it's just that np.add is not the correct reference for torch.add. It needs to be wrapped.

Reply:
Don't get hung up on splatting, if it's a bad idea or it would take a significant amount of time to figure out that's really not what this PR is about